src/robotFactory.js

```diff
@@ -1,10 +1,78 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
 
-class FlyingRobot {}
+    this.coords.x = coords.x
+      ? coords.x
+      : 0;
 
-class DeliveryDrone {}
+    this.coords.y = coords.y
+      ? coords.y
+      : 0;
+    this.chipVersion = chipVersion;
+  }
+  getInfo() {
+    const info = 'Robot: ' + this.name + ', Chip version: '
+    + this.chipVersion + ', Weight: ' + this.weight;
+
+    return info;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z
+      ? coords.z
+      : 0;
+  }
```

Comment:
Same issue as in the case above :) 
```suggestion
    const { x = 0, y = 0, z = 0 } = coords;

    this.coords = {
      x,
      y,
      z,
    };
```